pipeline with state tracking, launch scripts, docs

adamhrin-amd opened the pull request:
## Add Post-Training Evaluation Pipeline

Adds an automated pipeline for running instruction-following evaluations on chat models.

### Features

- **Pipeline script** (`posttraining_evals.sh`) that runs four evaluations:
  - IFEval - instruction following
  - MTBench - multi-turn conversations (inference + judge phases)
  - AlpacaEval - instruction quality with GPT-4 judge
  - Multi-IF - multi-turn instruction following

- **State tracking** via JSON file to resume interrupted runs and avoid duplicate jobs

### Usage

```bash
### Run all evaluations (default language en)
./posttraining_evals.sh /path/to/model

### Specific language
./posttraining_evals.sh /path/to/model fi

### Specific language and evaluations
./posttraining_evals.sh /path/to/model en ifeval mtbench
```

### Implementation
- Auto-clones required repositories (FastChat, alpaca_eval, Multi-IF)
- Creates isolated virtual environments for each evaluation
- Individual launch scripts for each evaluation type

Requires OPENAI_API_KEY and OPENAI_ORG_ID for MTBench and AlpacaEval judge phases.

dzautner commented on posttraining_evals.sh:
```diff
@@ -0,0 +1,576 @@
+    echo "Set it with: export OPENAI_ORG_ID=your_org_id_here"
+fi
+
+# Repository URLs
```
would be good to reference a specific commit/tag in each url so we run the same code every time

dzautner commented on posttraining/launch_alpaca_eval.sh:
```diff
@@ -0,0 +1,48 @@
+#SBATCH --exclusive=user
+#SBATCH --hint=nomultithread
+#SBATCH --gpus-per-node=mi250:8
+#SBATCH --account=project_462000353
```
would be good to paramertize this and other params and pass them downwards from from posttraining_evals.sh so it will be more portable, customisable

dzautner commented on posttraining/launch_mtbench_judge.sh:
```diff
@@ -0,0 +1,41 @@
+python gen_judgment.py \
+    --model-list "$MODEL_ID"-"$LANGUAGE" \
+    --parallel 2 \
+    --judge-model gpt-4o-2024-08-06 \
```
should probably also be parametrised?

dzautner commented on posttraining/launch_mtbench_inference.sh:
```diff
@@ -0,0 +1,52 @@
+module load pytorch/2.5
+
+# Check if FastChat venv exists, if not create it
+if [[ ! -d ".fastchat_venv" ]]; then
```
wonder can we avoid venv as they blow up the inode count on lumi

adamhrin-amd replied:
That is true, I would also prefer to avoid them but since each of the frameworks have their own set of requirements that seemed easiest to me to do it this way. Would you have a suggestion for how to avoid these?

adamhrin-amd commented:
I agree that it is not deeply integrated. This was mostly me running these evals separately from the evals framework and I thought that it might be useful to share if someone would want to run these quickly. But I agree that if we do have a plan to integrate these more deeply than this could be modified or completely discarded. I haven't done a deep exploration for what would be needed to integrate these

adamhrin-amd commented:
Added the result saving by parsing the results from the respective show_results helper scripts, or by parsing the output files in the respective workdirs. Also squashed all commit into one. Please review @dzautner @jonabur @ezosa

dzautner commented on evals/evals.py:
```diff
@@ -90,6 +90,57 @@ def __init__(self, name, n_samples=1):
+    def get_results_custom(self, json_data):
+        # Extract win rate from AlpacaEval results
+        if "results" in json_data and "length_controlled_winrate" in json_data["results"]:
+            return json_dat["results"]["length_controlled_winrate"]
```
typo? json_dat instea of json_data

adamhrin-amd replied:
thanks, fixed!

dzautner commented on templates/mtbench_judge.sh:
```diff
@@ -0,0 +1,155 @@
+    echo "OPENAI_API_KEY is not set"
+    exit 1
+fi
+if [ -z "$$JUDGE_MODEL" ]; then
```
accidental extra $?

adamhrin-amd commented on templates/alpaca_eval.sh:
```diff
@@ -0,0 +1,244 @@
+mkdir -p $PYTHONUSERBASE
+export PATH=$PATH:$PYTHONUSERBASE/bin
+
+# Clone/update alpaca_eval repository
```
I added the locking code to the templates
